src/main/java/edu/cornell/kfs/module/purap/service/impl/CuElectronicInvoiceHelperServiceImpl.java

```diff
@@ -1292,43 +1291,35 @@ private String buldErrorString(AutoPopulatingList<ErrorMessage> errorMessageList
     }
 
     @Override
-    protected void attachInvoiceXMLWithRejectDoc(ElectronicInvoiceRejectDocument eInvoiceRejectDocument, File attachmentFile, String noteText) {
+    protected void attachInvoiceXMLWithRejectDoc(
+            final ElectronicInvoiceRejectDocument eInvoiceRejectDocument, 
+            final File attachmentFile, 
+            final String noteText) {
         Note note = null;
         try {
             note = documentService.createNoteFromDocument(eInvoiceRejectDocument, noteText);
             // KFSCNTRB-1369: Can't add note without remoteObjectIdentifier
             note.setRemoteObjectIdentifier(eInvoiceRejectDocument.getNoteTarget().getObjectId());
-        } catch (Exception e1) {
+        } catch (final Exception e1) {
             throw new RuntimeException("Unable to create note from document: ", e1);
         }
 
-        String attachmentType = null;
-        BufferedInputStream fileStream = null;
-        try {
-            fileStream = new BufferedInputStream(new FileInputStream(attachmentFile));
-        } catch (FileNotFoundException e) {
-            LOG.error( "Exception opening attachment file", e);
-        }
-
         Attachment attachment = null;
-        try {
+        try (BufferedInputStream fileStream = new BufferedInputStream(new FileInputStream(attachmentFile))) {
+            final String attachmentType = null;
             attachment = attachmentService.createAttachment(eInvoiceRejectDocument.getNoteTarget(),
                     attachmentFile.getName(), INVOICE_FILE_MIME_TYPE, (int) attachmentFile.length(), fileStream,
                     attachmentType);
+        } catch (final FileNotFoundException e) {
+            LOG.error("Exception opening attachment file", e);
+        } catch (final IOException e) {
+            throw new RuntimeException("Unable to create attachment", e);
```

Comment:
Should I remove the catch IOException and just log on catch Exception "Unable to create attachmeht"?

Reply:
Yes, removing the IOException part and just using the Exception part to do the catching and logging is fine with me.

Reply:
updated

Reply:
I was recently testing ClamAV changes by trying to attach a file containing a virus - the EICAR virus scanning test file, so not really a virus, but one that would fail the AV check. When I tried this in dev and emergency, I was presented with a stack trace with the message "java.lang.IllegalArgumentException: file contents failed virus scan" and "this message will not appear in production environments".
Granted, in my case the "attachmentService.createAttachment()" method is being called from "KualiDocumentActionBase.insertBONote()" which is called from "CuDistributionOfIncomeAndExpenseAction.insertBONote()", not from this code.
But I'm curious, what is the behavior when this code encounters a failure to attach the file due to a failed virus check?
Does removing the IOException handling change that behavior?

Reply:
@dp65 The old version of this code was just catching all exceptions (including virus scan failures), logging the problematic exception, and then proceeding as normal but without the attachment. The new code is very similar but includes some cleanup items that KualiCo introduced (like using try-with-resources statements and logging file-not-found problems). Removing the "catch IOException" shouldn't be a problem if we intend to preserve the log-but-don't-throw functionality, especially since that "catch IOException" part was another KualiCo addition/cleanup.

Reply:
@cah292 , Sounds good.  I don't have an opinion about _throw_ versus _don't throw_, but I definitely want the _log_ part retained. Going forward, I'd like to be able gather metrics about how many or how often we're catching viruses, and being able to continue to search the logs for those events is key. This change doesn't seem to affect such logging, so I agree it shouldn't be a problem.  Thanks for your explanations.